src/core/server/logging/logging_config.ts

```diff
@@ -58,31 +57,23 @@ export const loggerSchema = schema.object({
 
 /** @public */
 export type LoggerConfigType = TypeOf<typeof loggerSchema>;
+
 export const config = {
   path: 'logging',
-  schema: legacyLoggingConfigSchema.extends({
+  schema: schema.object({
     appenders: schema.mapOf(schema.string(), Appenders.configSchema, {
       defaultValue: new Map<string, AppenderConfigType>(),
     }),
     loggers: schema.arrayOf(loggerSchema, {
       defaultValue: [],
     }),
-    root: schema.object(
-      {
-        appenders: schema.arrayOf(schema.string(), {
-          defaultValue: [DEFAULT_APPENDER_NAME],
-          minSize: 1,
-        }),
-        level: levelSchema,
-      },
-      {
-        validate(rawConfig) {
-          if (!rawConfig.appenders.includes(DEFAULT_APPENDER_NAME)) {
-            return `"${DEFAULT_APPENDER_NAME}" appender required for migration period till the next major release`;
-          }
-        },
-      }
-    ),
+    root: schema.object({
+      appenders: schema.arrayOf(schema.string(), {
+        defaultValue: [DEFAULT_APPENDER_NAME],
+        minSize: 1,
+      }),
```

Comment:
Not directly related to this PR, but I saw that `root.appenders` has `minSize: 1` and a default value of `['default']`, when the other loggers (`loggerSchema`) don't have a min size and have a default value of `[]`

```ts
root.appenders: schema.arrayOf(schema.string(), {
  defaultValue: [DEFAULT_APPENDER_NAME],
  minSize: 1,
}),

loggerSchema = schema.object({
  appenders: schema.arrayOf(schema.string(), { defaultValue: [] }),
});
```

Does that really make sense?  Specifying a logger with no appenders would have the same effect as setting the logger's `level` to `off`, but do we really want to have 2 ways to do the same thing? Shouldn't we have `loggerSchema.appenders` reproduce the same behavior as the root's?

Reply:
> Shouldn't we have loggerSchema.appenders reproduce the same behavior as the root's?

IMO it's an oversight. Let's add `minSize:1` as well. Because:

> Specifying a logger with no appenders would have the same effect as setting the logger's level to off

it might be a surprise for some users

Reply:
Actually I forgot the config class was enhancing the base schema. If a logger doesn't no specify any appender, it fallbacks to using the closest parent's appenders, which is the expected (and tested) behavior.